src/infrastructure/repositories/catogorie.repositorie.ts

```diff
@@ -0,0 +1,41 @@
+import { Category } from '@prisma/client';
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from '../prisma/prisma.service';
+
+import { ICategoryRepository } from '@/domain/repositories/category.repository';
+import {
+  UpdateCategoryDto,
+  AddCategoryDto,
+} from '@/presentation/category/category.dto';
+
+@Injectable()
+export class DatabaseCategoryRepository implements ICategoryRepository {
```

Comment:
The name of the file is `categorie-repositorie`, it must be category-repository, just to keep the pattern